packages/hub/indexing/document-context.js

```diff
@@ -429,8 +438,14 @@ module.exports = class DocumentContext {
         let data = related ? { type: related.type, id: related.id } : null;
         ensure(pristineDocOut, 'relationships')[field.id] = Object.assign({}, value, { data });
       }
+    } else if (value.links && value.links.related) {
+      let isCollection = field.fieldType === '@cardstack/core-types::has-many';
+      let data = await this.relatedLinks(qs.parse(value.links.related.split('/api?')[1]), isCollection);
```

Comment:
I think that be evaluating this inside of `DocumentContext` we actually end up materializing the result of the query in the index since we use `DocumentContext` to generate the `pristinedoc` column in the index DB (you should be able to confirm this by looking directly at the documents table)--which will likely be out of date since we are not invalidating models based on their appearance in queries that are relationships. I think you need to move the materialization of these relationships wholly outside of `DocumentContext` until we actually do support this type of invalidation.

Reply:
A test that would reveal this problem would be a test that has a computed that is based on a query based relationship. Changing the underlying models that effect the output of a query based relationship should invalidate a computed field that consumes a query based relationship. My guess is that such a test would fail if we are saving the materialized resources from a query based relationship in the index.

Reply:
Ahh yes, not sure why I didn't notice I was putting it in the index before. I'll write the test you suggested.